packages/types/src/envelope.ts

```diff
@@ -34,7 +34,19 @@ export type BaseEnvelope<
   EnvelopeItem extends BaseEnvelopeItem<{ type: string }>,
 > = [CommonEnvelopeHeaders & EnvelopeHeaders & Record<string, unknown>, EnvelopeItem[]];
 
-export type EventEnvelopeItem = BaseEnvelopeItem<{ type: 'event' | 'transaction' }, Event>;
+export type EventEnvelopeItem = BaseEnvelopeItem<
+  {
+    type: 'event' | 'transaction';
+    sample_rates?: Array<{
+      id: any;
+      rate: any;
+    }>;
+  },
+  // TODO: This is here because we currently have logic around serializing an event before inserting it
+  // into the envelope as we support both the store and envelope endpoints. Once that is removed, we can change
+  // this to only take an event.
+  Event | string
```

Comment:
It sucks how hacky the request model is, means that I have to either do a larger scale refactor or do this.

Might just bite the bullet and do the larger scale refactor.